### DISCUSSION TITLE
jspsych.js 'v.toLowerCase is not a function' error

### DISCUSSION THREAD

**Discussion by yuvalharr** [jspsych/jsPsych#1655] [Q&A] [answered]
My experiment worked perfectly, and then suddenly started showing this when the the first html-keyboard-response trial is executed:

_v.toLowerCase is not a function
    at jspsych.js:2308
    at Array.map (<anonymous>)
    at Object.listener_function [as fn] (jspsych.js:2308)
    at HTMLBodyElement.root_keydown_listener (jspsych.js:2236)_

When I remove all html-keyboard-response trials the experiment seems to work perfectly well.
The jsPsych.js version is 6.3.0, and it happens both when running locally on my computer and when running via cognition.run.
I tried anything I've thought about and still nothing. Even the slightest idea about what could be the problem would be extremely helpful.

Thanks!

**Comment by jodeleeuw** [comment]
What is the choices parameter set to in the html-keyboard-response trial?

**Reply by yuvalharr** [reply]
It's [32] or [68, 75]

**Comment by jodeleeuw** [comment, marked as answer]
We removed support for numeric key codes in 6.3.0, as the corresponding feature in browsers is considered deprecated.

Try replacing the values with the corresponding character representation.

**Reply by yuvalharr** [reply]
Yesss it works now!

Still trying to figure out how come it worked before. Maybe I was still loading the old html-keyboard plugin, but the problem is solved.

Thanks so much!

**Reply by kernel52** [reply]
a useful thread for those who need to update their keycodes. 
https://stackoverflow.com/questions/1772179/get-character-value-from-keycode-in-javascript-then-trim
